.github/workflows/lint.yaml

```diff
@@ -27,24 +27,12 @@ jobs:
       - name: Check commit message
         run: hack/check-commits.sh
 
-      - name: Setup Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: '1.21'
-
-      - name: Install dependencies
-        run: |
-          sudo apt install -y make
-          go install gotest.tools/gotestsum@latest
-          go install go.uber.org/mock/mockgen@v0.4.0
+      - name: Setup all dependencies
+        uses: ./.github/actions/setup-dependencies
 
-      - name: Check that generated files have been updated
+      - name: Check that generated files have been updated and that go.mod is tidy
         run: |
           make generate
-          git diff --exit-code
-
-      - name: Check that go.mod is tidy
-        run: |
```

Comment:
To handle this issue: https://github.com/flightctl/flightctl/actions/runs/8632458059/job/23663198421?pr=205#step:6:158

Seems like make generate massages the go.mod/sum for some reason in CI